Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,369 @@
+        # completed.  If you want to process results greedily as they come in,
+        # loop over asyncio.as_completed()
+        cve_data_list = await asyncio.gather(*tasks, return_exceptions=True)
+
+    for cve_data in cve_data_list:
+        try:
+            cve_cna = cve_data['containers']['cna']
+            cve_metrics = cve_cna['metrics']
+            # This data is extracted like this due to the location of the relevant information in the schema.
+            # mitre.org uses CVE schema 5.0, which is described in the link below
+            # https://github.com/CVEProject/cve-schema/tree/master/schema/v5.0
+            # The containers -> cna path contains vulnerability information like severity, which is documented
+            # under the metrics list.
+        except KeyError as e:
+            logger.debug(f"They key {e} couldn't be found in the CVE json data for CVE {cveid}.")
+            continue
+        for i, cve_metric in enumerate(cve_metrics):
+            try:
+                cve_affected_versions = cve_cna['affected'][i]['versions']
+            except KeyError as e:
+                logger.debug(f"They key {e} couldn't be found in the CVE json data for CVE {cveid}.")
+                continue
+
+            for key in cve_metric:
+                if re.search(r'[cC][vV][sS]{1,2}V\d', key) and \
+                   cve_cna['affected'][i]['product'] == "Kubernetes" and \
+                   cve_metric[key]['baseScore'] >= CVE_SEVERITY:
+
+                    for v in cve_affected_versions:
+                        vi = K8sVersionInfo()
+                        if 'lessThanOrEqual' in v:
+                            vi = vi.extract_version(v['lessThanOrEqual'].strip("v"))
+                            vi.patch += 1
+                        elif 'lessThan' in v:
+                            vi = vi.extract_version(v['lessThan'].strip("v"))
+                        if vi not in cfvs:
+                            cfvs.append(vi)
+                    break
+
+    return cfvs
+
+
+async def get_k8s_cluster_version(kubeconfig):
+    """Get the k8s version of the cluster under test.
+    """
+    cluster_config = await kubernetes_asyncio.config.load_kube_config(kubeconfig)
+
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        version_api = kubernetes_asyncio.client.VersionApi(api)
+        ret = await version_api.get_code()
+
+        version = K8sVersionInfo.extract_version(ret.git_version.strip("v"))
+        version.date = datetime.datetime.strptime(ret.build_date, '%Y-%m-%dT%H:%M:%SZ')
+
+        return version, cluster_config.current_context['name']
+
+
+def is_current_version(version, k8s_version_list, cve_version_list=[]):
+    """Filter the versions depending on the usage times set by the Standard and
+    the times set for CVE versions.
+    """
+    try:
+        if diff_months(version.date, datetime.datetime.now()) >= MINOR_VERSION_CADENCE:
+            return False
+        else:
```

Comment:
`else` after `return` is unnecessary; we can save a lot of indentation by omitting the `else`.

Reply:
Removed the `else`